collector/src/main/java/com/bitmovin/analytics/BitmovinAdAnalytics.kt

```diff
@@ -0,0 +1,226 @@
+package com.bitmovin.analytics
+
+import com.bitmovin.analytics.ads.*
+import com.bitmovin.analytics.data.AdSample
+import com.bitmovin.analytics.data.AdEventData
+import com.bitmovin.analytics.utils.Util
+
+class BitmovinAdAnalytics(var analytics: BitmovinAnalytics) {
+    private var activeAdBreak: AdBreak? = null
+    private var activeAdSample: AdSample? = null
+    private var adPodPosition: Int = 0
+    private var adStartupTimestamp: Long? = null
+    private var currentTime: Long? = null
+    private var beginPlayingTimestamp: Long? = null
+    private var isPlaying: Boolean = false
+    private val adManifestDownloadTimes: HashMap<String, Long> = hashMapOf()
+
+    fun onAdStarted(ad: Ad) {
+        if (!ad.isLinear) {
+            return
+        }
+
+        this.resetActiveAd()
+        this.activeAdSample = AdSample(ad = ad)
+        val activeAdSample = this.activeAdSample ?: return
+        this.currentTime = null
+        activeAdSample.adStartupTime = if (this.adStartupTimestamp != null) Util.getTimeStamp() - this.adStartupTimestamp!! else null
```

Comment:
This is a bit hard to read imo.. clearer would be if it's less concise.
Also with your null check the smartCast should remove the need for `!!`
Maybe we can move this to a method or something - not sure - but it is a bit hard to  follow all in one  line